tests/test_grammar.py

```diff
@@ -137,6 +137,23 @@
     ("str_quoted_too_many_1", "''a'", GrammarParseError),
     ("str_quoted_too_many_2", "'a''", GrammarParseError),
     ("str_quoted_too_many_3", "''a''", GrammarParseError),
+    ("str_quoted_trailing_esc_1", "'abc\\\\'", "abc\\"),
+    ("str_quoted_trailing_esc_2", "'abc\\\\\\\\'", "abc\\\\"),
+    ("str_quoted_no_esc_1", '"abc\\def"', "abc\\def"),
+    ("str_quoted_no_esc_2", '"abc\\\\def"', "abc\\def"),
```

Comment:
To make those tests easier to understand, can you use raw strings?
```suggestion
    ("str_quoted_no_esc_2", r'"abc\\def"', r"abc\def"),
```

Reply:
Done in 3457d0c

(note: one thing a bit annoying with raw strings is they don't "work" for strings that end with a `\` or contain an escaped quote of the same quote type as the enclosing quotes, e.g. `"abc\"def"` => this is why there still remain some non-raw strings)